crates/flox-rust-sdk/src/models/search.rs

```diff
@@ -0,0 +1,261 @@
+use std::collections::HashMap;
+use std::io::BufRead;
+use std::str::FromStr;
+
+use flox_types::catalog::System;
+use flox_types::stability::Stability;
+use serde::{Deserialize, Serialize};
+use serde_json::Value;
+
+use crate::nix::flake_ref::FlakeRef;
+
+#[derive(Debug, thiserror::Error)]
+pub enum SearchError {
+    #[error("failed to deserialize search result from JSON: {0}")]
+    Deserialize(#[from] serde_json::Error),
+    #[error("couldn't parse stdout to separate JSON lines: {0}")]
```

Comment:
Is it normal to implicitly convert `std::io::Error` to something with a specific description like this? Seems like it would be easy to accidentally convert to this error from a completely unrelated `std::io::Error`

```suggestion
    #[error("couldn't split stdout into individual lines: {0}")]
```

Reply:
Sure it's possible to accidentally catch all `std::io::Error`s this way, but the only I/O we're doing is parsing `stdout` in this one spot so I don't see an issue yet. We can change this later if we ever do other I/O operations.

Reply:
I didn't actually answer the first part of the question. Yes, it's common to do this. If there's more than one failure mode that could generate an I/O error then you wouldn't do the automatic conversion.